File: TestP1/src/irrgarten/Controller.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

package irrgarten;



public class Controller{
    
    private Game game;
    private TextUI view;
    
    
    public Controller(Game game, TextUI view) {
        this.game = game;
        this.view = view;
    }
    
    public void play() {
        
       
        
    boolean endOfGame = false;
            
    //Modo de interfaz de texto
            
            
            
    while (!endOfGame) {
        
        view.showGame(game.getGameState()); 
        Directions direction = view.nextMove(); 
        //System.out.println(direction);        //DEPURACION
        endOfGame = game.nextStep(direction);
        }
            
        view.showGame(game.getGameState());    
    }
    
}

File: Irrgarten_ruby/controller.rb
# frozen_string_literal: true

module Irrgarten

  class Controller
    def initialize(game,view)
      @game = game
      @view = view
    end

    def play
      end_of_game = false
      while (!end_of_game)
        @view.show_game(@game.get_game_state)
        direction = @view.next_move
        end_of_game = @game.next_step(direction)
      end
      @view.show_game(@game.get_game_state)
    end
  end # class
end # module
